server/src/main/java/org/elasticsearch/search/aggregations/AggregationPhase.java

```diff
@@ -29,30 +31,49 @@ public static void preProcess(SearchContext context) {
         if (context.aggregations() == null) {
             return;
         }
-        BucketCollector bucketCollector;
-        try {
-            context.aggregations().aggregators(context.aggregations().factories().createTopLevelAggregators());
-            bucketCollector = MultiBucketCollector.wrap(true, List.of(context.aggregations().aggregators()));
-            bucketCollector.preCollection();
-        } catch (IOException e) {
-            throw new AggregationInitializationException("Could not initialize aggregators", e);
-        }
-        final Collector collector;
-        if (context.aggregations().factories().context() != null
-            && context.aggregations().factories().context().isInSortOrderExecutionRequired()) {
-            TimeSeriesIndexSearcher searcher = new TimeSeriesIndexSearcher(context.searcher(), getCancellationChecks(context));
-            searcher.setMinimumScore(context.minimumScore());
-            searcher.setProfiler(context);
+        final Supplier<BucketCollector> bucketCollectorSupplier = () -> {
             try {
-                searcher.search(context.rewrittenQuery(), bucketCollector);
+                Aggregator[] aggregators = context.aggregations().factories().createTopLevelAggregators();
+                context.aggregations().aggregators(aggregators);
+                BucketCollector bucketCollector = MultiBucketCollector.wrap(true, List.of(aggregators));
+                bucketCollector.preCollection();
+                return bucketCollector;
             } catch (IOException e) {
-                throw new AggregationExecutionException("Could not perform time series aggregation", e);
+                throw new AggregationInitializationException("Could not initialize aggregators", e);
             }
-            collector = BucketCollector.NO_OP_COLLECTOR;
+        };
+        final Supplier<Collector> collectorSupplier;
+        if (context.aggregations().isInSortOrderExecutionRequired()) {
+            executeInSortOrder(context, bucketCollectorSupplier.get());
+            collectorSupplier = () -> BucketCollector.NO_OP_COLLECTOR;
```

Comment:
Do we still need to register the collector manager when the collector is a no-op one?

Reply:
I am following the current code and I assume that if we were adding that before is because it is needed. Maybe as a follow up we can just remove it and see if anything breaks.

Reply:
agreed, no need to change it now.